scripts/bootstrap.sh

```diff
@@ -119,7 +119,7 @@ vcpkgDownloadFile()
     url=$1; downloadPath=$2 sha512=$3
     vcpkgCheckRepoTool "curl"
     rm -rf "$downloadPath.part"
-    curl -L $url --tlsv1.2 --create-dirs --retry 3 --output "$downloadPath.part" || exit 1
+    curl -L $url --tlsv1.2 --create-dirs --retry 3 --output "$downloadPath.part" --silent --show-error || exit 1
```

Comment:
```suggestion
    curl -L $url --tlsv1.2 --create-dirs --retry 3 --output "$downloadPath.part" --silent --show-error --fail || exit 1
```
Is `--fail` appropriate here?

Reply:
Hrm good question - `--silent --show-error` was my suggestion, let me research what `--fail` does.

Reply:
I'm hesitant to suggest the use of `--fail`. The `curl` docs say:

> `-f, --fail          Fail silently (no output at all) on HTTP errors`

The reason for `--silent --show-error` is to get no output except in the case of errors. `--silent` means no output, and `--show-error` means show errors even in silent mode.

Reply:
@BillyONeal says it changes "http 404 + here's a document explaining that it was a 404" into "no file and return an error code", so it seems good.